packages/components/Button/src/Button.android.tsx

```diff
@@ -0,0 +1,82 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { Pressable, View } from 'react-native';
+import { IButtonSlotProps, IButtonState, IButtonProps, IButtonRenderData, buttonName, IButtonType } from './Button.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { Text } from '@fluentui-react-native/text';
+import { settings } from './Button.settings';
+import { backgroundColorTokens, borderTokens, textTokens, foregroundColorTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
+import { filterViewProps } from '@fluentui-react-native/adapters';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { useAsPressable, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { Icon } from '@fluentui-react-native/icon';
+import { createIconProps } from './Button.icon';
+
+export const Button = compose<IButtonType>({
+  displayName: buttonName,
+  usePrepareProps: (userProps: IButtonProps, useStyling: IUseComposeStyling<IButtonType>) => {
+    const { icon, content, accessibilityLabel = userProps.content, testID, onClick, ...rest } = userProps;
+    // attach the pressable state handlers
+    const pressable = useAsPressable({ ...rest, onPress: onClick });
+    // set up state
+    const state: IButtonState = {
+      info: {
+        ...pressable.state,
+        disabled: !!userProps.disabled,
+        content: !!content,
+        icon: !!icon,
+      },
+    };
+
+    const buttonRef = useViewCommandFocus(userProps.componentRef);
+    // grab the styling information, referencing the state as well as the props
+    const styleProps = useStyling(userProps, (override: string) => state.info[override] || userProps[override]);
+    // create the merged slot props
+
+    const slotProps = mergeSettings<IButtonSlotProps>(styleProps, {
+      ripple: {
+        ...pressable.props,
+        ref: buttonRef,
+        accessibilityLabel: accessibilityLabel,
+        accessibilityState: { disabled: state.info.disabled },
+        focusable: !state.info.disabled,
+      },
+      content: { children: content, testID: testID },
+      icon: createIconProps(icon),
+    });
+
+    return { slotProps, state };
+  },
+  settings,
+  render: (Slots: ISlots<IButtonSlotProps>, renderData: IButtonRenderData, ...children: React.ReactNode[]) => {
+    const info = renderData.state!.info;
+    return (
+      <Slots.root>
+        <Slots.ripple>
+          <Slots.stack>
+            {info.icon && <Slots.icon />}
+            {info.content && <Slots.content />}
+            {children}
+          </Slots.stack>
+        </Slots.ripple>
+      </Slots.root>
+    );
+  },
+  slots: {
+    root: View,
+    ripple: Pressable,
```

Comment:
When the button has Pressable as root, the ripple effect overflows. 

![Pressable-ripple-leaks](https://user-images.githubusercontent.com/23552252/118124829-c35d9300-b413-11eb-8f61-7431dea5e3bc.gif)

Reply:
I'm not sure I see the overflow in that video?

Reply:
Also, does this comment thread give any insight in general? https://github.com/microsoft/fluentui-react-native/pull/467/files#r499365330

Reply:
>I'm not sure I see the overflow in that video?

If you look at the corners, you will see some translucent layer flowing out.

>Also, does this comment thread give any insight in general? https://github.com/microsoft/fluentui-react-native/pull/467/files#r499365330

Yep, started this thread in response to the argument that a single wrapper View could be used or not. So I began with just using Pressable as the root view, but noticed that ripple flows outside of the border radius, and the workaround was to wrap Pressable with a View and apply styles to hide it.